ci/utils/ubuntu/run_smoke_test.sh

```diff
@@ -74,14 +89,38 @@ verifytd() {
         ssh-keygen -f "$home_dir/.ssh/known_hosts" -R "[localhost]:$TD_GUEST_PORT"
     fi
     
-    out=$(sshpass -p "$TD_GUEST_PASSWORD" ssh -o StrictHostKeyChecking=no -p "$1" root@localhost 'dmesg | grep -i tdx' 2>&1)
+    IMAGE_TYPE=$3
+    if [ "$IMAGE_TYPE" == "custom" ]; then
+	echo "This is custom image hence setting the guest user and password"
+	TD_GUEST_PASSWORD=intel123
+	TD_GUEST_USER=tdx_test
+    fi
+    out=$(sshpass -p "$TD_GUEST_PASSWORD" ssh -o StrictHostKeyChecking=no -p "$1" $TD_GUEST_USER@localhost 'sudo dmesg | grep -i tdx' 2>&1)
     echo "$out"
     if [[ "$out" =~ "${TD_GUEST_VERIFY_TEXT}" ]]; then
-        echo "td guest is configured"
+	echo "td guest is configured (checked using port number)"
+    else
+	echo "td guest is not configured (checked using port number)"
+        exit 1
+    fi
+
+    ip_addr=$(sshpass -p "$TD_GUEST_PASSWORD" ssh -o StrictHostKeyChecking=no $TD_GUEST_USER@"$2" 'sudo dmesg | grep -i tdx' 2>&1)
+    echo "$ip_addr"
+    if [[ "$ip_addr" =~ "${TD_GUEST_VERIFY_TEXT}" ]]; then
+	echo "td guest is configured (checked using ip address)"
     else
-        echo "td guest is not configured"
+        echo "td guest is not configured (checked using ip address)"
         exit 1
     fi
+
+    username=$(sshpass -p "$TD_GUEST_PASSWORD" ssh -o StrictHostKeyChecking=no -p "$1" $TD_GUEST_USER@localhost 'whoami' 2>&1)
```

Comment:
Code to confirm the VM username matches the one provided during create-td-image.